internal/querynode/historical.go

```diff
@@ -159,7 +159,48 @@ func (h *historical) search(searchReqs []*searchRequest, collID UniqueID, partID
 	return searchResults, searchSegmentIDs, searchPartIDs, err
 }
 
-// getSearchPartIDs fetchs the partition ids to search from the request ids
+// validateSegmentIDs checks segments if their collection/partition(s) have been released
+func (h *historical) validateSegmentIDs(segmentIDs []UniqueID, collectionID UniqueID, partitionIDs []UniqueID) (validatedSegmentIDs []UniqueID, err error) {
+	// validate partitionIDs
+	validatedPartitionIDs, err := h.getTargetPartIDs(collectionID, partitionIDs)
+	if err != nil {
+		return
+	}
+	log.Debug("search validated partitions", zap.Int64("collectionID", collectionID), zap.Int64s("partitionIDs", validatedPartitionIDs))
+
+	col, err := h.replica.getCollectionByID(collectionID)
+	if err != nil {
+		return
+	}
+
+	// return if no partitions are loaded currently
+	if len(validatedPartitionIDs) == 0 {
+		switch col.getLoadType() {
+		case loadTypeCollection:
+			err = fmt.Errorf("partitions have been released, collectionID = %d, target paritition= %d", collectionID, partitionIDs)
+		case loadTypePartition:
+			err = fmt.Errorf("collection has been released, collectionID = %d, target paritition= %d", collectionID, partitionIDs)
```

Comment:
NIT: `partitionIDs` is a slice. Use %v in format string.

Reply:
fixed